### ISSUE TITLE
Set properties to uploaded blob (eg. ContentType)

### ISSUE THREAD

**Issue by heyarny** [Azure/azure-storage-java#437]
I'm really having trouble while using this SDK.
How do I set properties to the blob I'm uploading? There's just Metadata I can set..

**Comment by heyarny** [comment]
Wasted a whole day to search for that option to figure out it's not even implemented yet?
Please rename this SDK to "Preview" or something. I do not recommend to use this SDK at all!

Switched to legacy sdk and wrote the same case with half the lines.

**Comment by rickle-msft** [comment]
Hi, @heyarny.  Thank you for your feedback. What properties are you trying to set exactly? In addition to metadata, you can pass BlobHTTPHeaders to any method that creates a blob. This allows you to set fields such as content type, content encoding, etc. Which fields are you seeing as not implemented?

**Comment by heyarny** [comment]
@rickle-msft this is what I was looking for, but all the REST documentation is talking about Blob Properties. Metadata are headers as well, but you don't call them "headers".
Why don't you stay lined with the general REST API documentation and keep the names to not confuse developers with new naming-conventions/redefinitions.

- REST API = "blob properties"
- C# = "blob properties"
- Python = "blob properties"
- JavaScript = "blob headers"?
- Java = "blob headers"?

A more unified concept of all SDKs would be a smarter way to go than one official REST API and SDKs with 10 different concepts. Not sure whether the "reactivex" way is the smarter way to go, because it will be completely different to all other languages. Keep it simple!

**Comment by heyarny** [comment]
Oh, and not to forget:
on SDK8 it was "blob properties" -> BlobProperties.
SDK10 it's BlobHTTPHeaders all of a sudden?
Not even google could help me to find this one.

Staying on SDK8. Thanks anyway.

**Comment by rickle-msft** [comment]
@heyarny. I understand you are staying on v8, but I will respond to your concerns for completeness. 

The aforementioned set of properties was renamed to BlobHTTPHeaders because those are not an Azure Storage concept, but rather an HTTP concept. The renaming was an intentional departure from the idea of simply BlobProperties because the SetProperties REST api actually does more than just set these few fields, and we wanted to make it more clear that only the HTTP-related properties would be affected by that method. 

There actually is a significant internal effort to unify the SDKs across languages that is underway right now, of which Java v10 is one part. [Go](https://github.com/Azure/azure-storage-blob-go/blob/master/azblob/url_blob.go#L119) looks similar as well, as does the latest version of JavaScript. Other languages will likely be following suit. The discrepancies you are seeing are some growing pains associated with the fact that we are unable to release all the new versions at once. We acknowledge that it is not ideal that there are inconsistencies, but there is not much we can do to help that while the other sdks are in development. 

Your feedback on reactivex is likewise noted. The new effort is trying to have all the new versions of SDKs be asynchronous, and it seemed from our research that Rx was the preferred async solution for the Java community.  Would you prefer the sdk was simply entirely sync, or do you think there is a better option for offering async behavior in Java?

I am sorry again that you experienced this frustration. In the future, please feel free to post here with such questions before feeling like you have wasted a large amount of time. We will be happy to work with you as best we can and hopefully unblock you before you get frustrated. 

I will leave this open for a few days if you would like to offer feedback on the information I have provided as we are in the process of evaluating lots of feedback right now. Otherwise, I will close this issue if I have not heard from you in a few days as I believe your actual problem has been addressed.
